books/serializers.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework import serializers
+
+from books.models.books import Book, Author, Publisher
+from books.models.country import Country
+
+
+class BookSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    publisher = serializers.SlugRelatedField(slug_field='name', read_only=True)
+    country = serializers.SlugRelatedField(slug_field='name', read_only=True)
+    authors = serializers.SlugRelatedField(slug_field='name', many=True, read_only=True)
+
+    class Meta:
+        model = Book
+        fields = ("id", "name", "isbn", "authors", "number_of_pages", "publisher", "country", "release_date")
+
+    def to_representation(self, instance):
+        data = super(BookSerializer, self).to_representation(instance)
+
+        return data
+
+    def create(self, validated_data):
+        request = self.context['request']
+        country, _ = Country.objects.get_or_create(name=request.data.get('country'))
```

Comment:
@xhahid43eb Creating objects on the fly without any validations isn't a good idea, you will end up with invalid country and publisher names in your database. Create/populate publisher from external source and only assign their id's here while creating book instance.

Reply:
I do this because of the document requirement otherwise more easier to manage with  populated data